### DISCUSSION TITLE
Compiles and Runs from Source on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by TahomaSoft** [olive-editor/olive#2016] [Show and tell]
Hi all,

Just discovered Olive, and got it to compile and run on Ubuntu 22 with only minor deviations from the published instructions for Ubuntu 20.

**Comment by Simran-B** [comment]
Cool, mind sharing what needs to be done differently?

**Comment by TahomaSoft** [comment]
From memory (I didn't write down the steps for this first go, but can when I do it again on another machine):
A few of the dependencies (OpenColor or OpenEXR) didn't install with the exact command given, but a little apt-cache search provided me with the similar names for the packages I needed to install. I think there was also a build failure for a dependency that I was able to solve the same way. 

I also made the mistake of adding OpenColor v1 before trying to compile Olive (which was NOT in the Olive manual); I had to remove it and compile and install OpenColor v2 as directed. 

BTW, this is much easier to compile than DigiKam, for example.  I was very pleasantly surprised how well the compile went. I can also try on a recent (35, but not 36, yet) version of Fedora.

---Edit---
Or, more topically, the FOSS video editor Cinelerra or the non-free DaVinci Resolve. Never been able to get either of them to compile and/or run on Linux.

**Comment by TahomaSoft** [comment]
And I was able to get it to lightly edit and render a 5K video (5120 × 2880).

**Comment by alexmyczko** [comment]
or you could go with the official packaging: https://launchpad.net/ubuntu/+source/olive-editor

some use shotcut, it is also a great video editor

**Comment by JoseOlin** [comment]
Hey, I would also be interested in knowing how to compile on Ubuntu 22, I use Linux Mint 21.1 which is based on it.

I would love to contribute, at least as a tester on my platform, and the appImage sometines does not provide enough information to trace the problem.

I remember compiling on Ubuntu 18 was very easy but on 22 it still gives me problems finding Imath, OpenTimeIO and GoogleCrashpad.

<details><summary>Details</summary>
<p>

-- Could NOT find Imath (missing: Imath_DIR)
-- Found CONFIG for OpenEXR 2 (OPENEXR_VERSION=2.5.7)
-- Could NOT find OpenTimelineIO (missing: OTIO_LIBRARIES OTIO_INCLUDE_DIRS OTIO_DEPS_INCLUDE_DIR) 
   OpenTimelineIO interchange will be disabled.
-- Could NOT find GoogleCrashpad (missing: CRASHPAD_CLIENT_LIB CRASHPAD_COMMON_LIB CRASHPAD_UTIL_LIB CRASHPAD_BASE_LIB BREAKPAD_BIN_DIR CRASHPAD_BUILD_INCLUDE_DIR CRASHPAD_CLIENT_INCLUDE_DIR CRASHPAD_BASE_INCLUDE_DIR CRASHPAD_COMPAT_LIB) 
   Automatic crash reporting will be disabled.
CMake Error at ext/CMakeLists.txt:17 (add_subdirectory):
  The source directory

    /media/Ext4/Programs/Olive-editor/olive-master_23-05/olive/ext/core

  does not contain a CMakeLists.txt file.


CMake Error at ext/CMakeLists.txt:21 (add_subdirectory):
  The source directory

    /media/Ext4/Programs/Olive-editor/olive-master_23-05/olive/ext/KDDockWidgets

  does not contain a CMakeLists.txt file.


-- Configuring incomplete, errors occurred!

</p>
</details>
